### PR TITLE
Live playlist loading improvements

### PR DESCRIPTION
**Pull request by ntadej**
We managed to fix the problems described in #236. This PR consists of two commits.

The repetition of chunks was caused because media index didn't need to be translated when changing quality. The playlist was the same and translation should not occur.

The second commit fixes crazy loading of playlists and chunks after several quality changes as timeouts accumulated and produced more and more requests.

### CODE REVIEW

File: src/playlist-loader.js [hunk 1]
```diff
@@ -94,6 +94,7 @@
 
           // refresh live playlists after a target duration passes
           if (!loader.media().endList) {
+            window.clearTimeout(mediaUpdateTimeout);
```
Comment (dmlap) [inline src/playlist-loader.js]: This had to be a huge pain to track down. Thank you!

File: src/videojs-hls.js [hunk 2]
```diff
@@ -1030,6 +1030,11 @@ videojs.Hls.translateMediaIndex = function(mediaIndex, original, update) {
     // let the media index be zero when there are no segments defined
     return 0;
   }
+  
+  if (original.mediaSequence === update.mediaSequence) {
+	// playlist sequence didn't change
+	return mediaIndex;
```
Comment (dmlap) [inline src/videojs-hls.js]: Isn't this equivalent to the [computation of translatedMediaIndex](https://github.com/ntadej/videojs-contrib-hls/blob/playlist-fix/src/videojs-hls.js#L1041)? If `original.mediaSequence === update.mediaSequence`, then:

```
(mediaIndex + (original.mediaSequence - update.mediaSequence))
=> (mediaIndex + (0))
=> mediaIndex
```
The only case where that value is unused is if the resulting index is outside the scope of the current playlist. Were you seeing situations where `translatedMediaIndex === update.segments.length`?

Comment (ntadej) [inline src/videojs-hls.js]: Yes, the problem is when `translatedMediaIndex === update.segments.length`. The index gets increased and falls out of the playlist.

Comment (dmlap) [inline src/videojs-hls.js]: Ok, that makes sense. It seems to me that the conditional is wrong. If we made the test `translatedMediaIndex >= update.segments.length + 1`, would that resolve the issue?

**Comment by ntadej** [comment]
Updated PR according to your comments.

File: src/videojs-hls.js [hunk 3]
```diff
@@ -1035,7 +1035,7 @@ videojs.Hls.translateMediaIndex = function(mediaIndex, original, update) {
   // bitrate switches should be happening here.
   translatedMediaIndex = (mediaIndex + (original.mediaSequence - update.mediaSequence));
 
-  if (translatedMediaIndex >= update.segments.length || translatedMediaIndex < 0) {
+  if (translatedMediaIndex > update.segments.length || translatedMediaIndex < 0) {
```
Comment (bc-bbay) [inline src/videojs-hls.js]: This change doesn't feel right. Segments is a 0-indexed array, so update.segments.length is an out-of-bounds element. Why is this necessary?

Comment (dmlap) [inline src/videojs-hls.js]: When all the segments currently available in a live playlist have been downloaded, mediaIndex will point to one beyond the end. There's nothing wrong with this situation, we're just waiting for a playlist refresh. It's incorrect to reset mediaIndex back to the live point if this occurs.
